### ISSUE TITLE
Unknown rule in disable directive throwing unnecessary errors

### ISSUE THREAD

**Issue by jbmanning** [eslint/eslint#14851]
**The version of ESLint you are using.**
v7.31.0

**The problem you want to solve.**
I have two eslint configs, one for stylistic auto-fix's and one for linting. I am running into a problem when I have a disable directive (eg. `// eslint react/no-danger: "off"`) which includes a rule that is only available in one of the configs. I get errors of `Definition for rule '...' was not found` because that plugin has not been included in both config files. 

**Your take on the correct solution to problem.**
I'd propose three solutions: 
1. Include these cases under the existing config flag `--report-unused-disabled-directives`, since these are effectively just unused directives. 
2. Add a config flag `--report-unknown-disable-directives` which disables, or changes the reporting mechanism for these cases.
3. Unknown rules in disable directives should report as a warning, rather than an error.

**Are you willing to submit a pull request to implement this change?**
Yes, with minor guidance

**Comment by nzakas** [comment]
This might actually be a bug. When rules are set to off in a config file, we purposely don’t check if the rule exists to avoid throwing an error. It seems like we should do the same with disable directives.

I’d like some additional opinions from the team on this.

**Comment by mdjermanovic** [comment]
This is intended behavior (https://github.com/eslint/eslint/issues/9505), introduced in ESLint v6.0.0

**Comment by jbmanning** [comment]
Reading through that thread it sounds like there was some hesitance around the change due to issues like above, as well as shared config files with disabled rules that don't necessarily exist. 

I would still feel comfortable with any of my above recommendations if anyone had thoughts on that, or reconsideration about the linked behavior introduced in v6?

**Comment by nzakas** [comment]
@mdjermanovic I think that was before we had the ability to report unused disable directives and seems inconsistent with how config files work. It seems like [we decided](https://github.com/eslint/eslint/issues/9505#issuecomment-491055516) that different behavior for disable directives and config files was undesirable but didn’t have a good solution.

I do wonder if we should revisit that decision now that we have a way to report unused disable directives.

**Comment by nzakas** [comment]
@eslint/eslint-tsc do we still want to throw errors in disable directives if they disable a rule that doesn’t exist? We don’t do this in config files and I’d prefer that disable directives also not throw.

**Comment by mdjermanovic** [comment]
> two eslint configs, one for stylistic auto-fix's and one for linting

I think the main problem here is that eslint directive comments do not support this use case.

Aside from `/* eslint react/no-danger: "off" */`. any of these would also produce linting errors:

```js
/* eslint react/no-danger: "warn" -- changes severity for this file */

/* eslint react/no-danger: ["error", "something"] -- reconfigures rule for this file */

// eslint-disable-line react/no-danger

// eslint-disable-next-line react/no-danger

/* eslint-disable react/no-danger */

/* eslint-enable react/no-danger */

```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHJlYWN0L25vLWRhbmdlcjogXCJ3YXJuXCIgLS0gY2hhbmdlcyBzZXZlcml0eSBmb3IgdGhpcyBmaWxlICovXG5cbi8qIGVzbGludCByZWFjdC9uby1kYW5nZXI6IFtcImVycm9yXCIsIFwic29tZXRoaW5nXCJdIC0tIHJlY29uZmlndXJlcyBydWxlIGZvciB0aGlzIGZpbGUgKi9cblxuLy8gZXNsaW50LWRpc2FibGUtbGluZSByZWFjdC9uby1kYW5nZXJcblxuLy8gZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIHJlYWN0L25vLWRhbmdlclxuXG4vKiBlc2xpbnQtZGlzYWJsZSByZWFjdC9uby1kYW5nZXIgKi9cblxuLyogZXNsaW50LWVuYWJsZSByZWFjdC9uby1kYW5nZXIgKi9cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTMsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

This isn't related to `--report-unused-disable-directives`, we're reporting all these directives as invalid (due to unknown rule id), not as unused. I believe this behavior is desirable for most users - it seems better to report a possible error (e.g., typo) than to silently ignore the directive - even if the directive aims to set a rule to `"off"`. 

Special-casing `"unknown/rule": "off"` would solve just a small part of this problem, a complete solution would be something like the one proposed in #13472.

**Comment by nzakas** [comment]
So let’s circle back, do we want to make a change here or stick with the current behavior?
